File: python/rotational-cipher/rotational_cipher.py
def rotate(text, key):
    """
    Encrypt a given text value by replacing each character with another 
    character that is key spaces away in the alphabet. If you go past Z
    you should wrap around to A again. This is case sensitive so we will handle
    A-Z, and a-z separately. Characters that are not in the alphabet will be
    passed through to the output unchanged.

    For example ABC rotate 4 is EFG. 

    parameters:
    text: string, the value to encrypt
    key: number, is the offset
    """
    cipher = ''         # return value
    capA = ord('A')     # code for capital A (only want to compute once)
    capZ = ord('Z')     # code for captial Z
    lowerA = ord('a')   # code for lower case A
    lowerZ = ord('z')   # code for lower case Z

    for ch in text:
        code = ord(ch)
        if code >= capA and code <= capZ:
            code = code + key
            while code > capZ:
                code -= (capZ - capA + 1)
            cipher += chr(code)            
        elif code >= lowerA and code <= lowerZ:
            code = code + key
            while code > lowerZ:
                code -= (lowerZ - lowerA + 1)
            cipher += chr(code)
        else:
            cipher += ch
    
    return cipher

File: php/rotational-cipher/RotationalCipher.php
<?php
// Rotational (Caeser) Cipher exercise.
// Note to self, I should have used array_map

declare(strict_types=1);

const ALPHABET_SIZE = 26;

class RotationalCipher
{
    // Rotate a single character by a shift for a given start of the alphabet
    // @param $char: The character to rotate
    // @param $minChar: The character at the start of the alphabet ("a", or "A")
    // @param $shift: How many places to rotate the character.
    // @returns: The rotated character
    private function rotateChar(string $char, string $minChar, int $shift): string
    {
        $index = ord($char) - ord($minChar);
        $index += $shift;
        $index = $index % ALPHABET_SIZE;
        return chr($index + ord($minChar));
    }
    
    // Rotate a text string by a set number of places.
    // Characters that are not A-Z or a-z will not be rotated and will propogate to the
    // output as-is.
    // @param $text: The text to rotate
    // @param $shift: The number of places to rotate each character.
    // @returns: Copy of $text but with the alphabetic characters rotated by the shift amount.
    public function rotate(string $text, int $shift): string
    {

        $result = "";
        foreach(str_split($text, 1) as $char) {
            if ($char >= "a" && $char <= "z") {
                $result .= $this->rotateChar($char, "a", $shift);
            } else if ($char >= "A" && $char <= "Z") {
                $result .= $this->rotateChar($char, "A", $shift);
            } else {
                $result .= $char;
            }
        }
        return $result;    
    }
}
